packages/fuels-programs/src/call_utils.rs

```diff
@@ -324,14 +330,28 @@ pub(crate) fn get_single_call_instructions(
 
     let mut instructions = [
         op::movi(0x10, call_data_offset),
-        op::movi(0x11, gas_forwarded_offset),
+        op::movi(0x11, amount_offset),
         op::lw(0x11, 0x11, 0),
-        op::movi(0x12, amount_offset),
-        op::lw(0x12, 0x12, 0),
-        op::movi(0x13, asset_id_offset),
-        op::call(0x10, 0x12, 0x13, 0x11),
+        op::movi(0x12, asset_id_offset),
     ]
     .to_vec();
+
+    match offsets.gas_forwarded_offset {
+        Some(gas_forwarded_offset) => {
+            let gas_forwarded_offset = gas_forwarded_offset
+                .try_into()
+                .expect("gas_forwarded_offset out of range");
```

Comment:
At some point in the future, we should probably consider changing `offets.gas_forwarded_offset` and its brethren to `u32` to fit the instruction.